camayoc/tests/remote/quipucords/test_scan.py

```diff
@@ -0,0 +1,240 @@
+"""Tests for  quipucords scans and reports.
+
+These tests are parametrized on the inventory listed in the config file.
+
+:caseautomation: automated
+:casecomponent: api
+:caseimportance: high
+:caselevel: integration
+:requirement: quipucords
+:testtype: functional
+:upstream: yes
+"""
+
+import pytest
+
+from camayoc import api, utils
+from camayoc.qcs_models import (
+        Credential,
+        Source,
+        Scan
+)
+from camayoc.config import get_config
+from camayoc.constants import (
+    VCENTER_CLUSTER,
+    VCENTER_DATA_CENTER,
+    VCENTER_HOST,
+)
+from camayoc.exceptions import ConfigFileNotFoundError
+from camayoc.tests.remote.utils import wait_until_live
+from camayoc.tests.qcs.api.v1.utils import wait_until_state
+
+
+SCAN_REPORTS = {}
+"""Cache to associate the report ids with the scans named in config file."""
+
+@pytest.fixture(scope='module')
+def cleanup():
+    """Fixture that cleans up any created host credentials."""
+    trash = []
+
+    yield trash
+
+    for obj in trash:
+        obj.delete()
+
+def create_cred(cred_info, cleanup):
+    """Given info about cred from config file, create it and return id. """
+    c = Credential(
+        cred_type=cred_info['type'],
+        username=cred_info['username'],
+    )
+    if cred_info.get('sshkeyfile'):
+        c.ssh_keyfile = cred_info['sshkeyfile']
+    else:
+        c.password = cred_info['password']
+    c.create()
+    cleanup.append(c)
+    return c._id
+
+def create_source(source_info,cred_name_to_id_dict, cleanup):
+    """Given info about source from config file, create it and return id."""
+    cred_ids = [value for key,value in cred_name_to_id_dict.items() if key in source_info['credentials']]
+    host = source_info.get('ipv4') if source_info.get('ipv4') else source_info.get('hostname')
+    src = Source(
+        source_type=source_info.get('type', 'network'),
+        hosts=[host],
+        credential_ids=cred_ids,
+        options=source_info.get('options')
+    )
+    src.create()
+    cleanup.append(src)
+    return src._id
+
+def run_scan(src_ids, scan_type='inspect'):
+    """Scan a machine and return its report id.
+
+    If the scan fails to generate a report, an AssertionError will be raised.
+    """
+    scan = Scan(source_ids=src_ids, scan_type=scan_type)
+    scan.create()
+    wait_until_state(scan, timeout=600, state='completed')
+    report_id = scan.results().json()
+    return report_id
+
+@pytest.fixture(scope='module', autouse=True)
+def run_all_scans(vcenter_client, cleanup):
+    """Run all configured scans caching the report id associated with each.
+
+    Run each scan defined in the ``qcs`` section of the configuration file.
+
+    Cache the report id of the scan, associating it with its scan.
+
+    The expected configuration in the Camayoc's configuration file is as
+    follows::
+
+        qcs:
+        # other needed qcs config data
+        #
+        # specific for remote scans:
+            - remote_scans:
```

Comment:
I would name this `scans` because we have `sources` to map sources.

Reply:
good idea